src/main/java/com/blamejared/controlling/client/gui/GuiNewControls.java

```diff
@@ -216,9 +220,16 @@ public void drawScreen(int mouseX, int mouseY, float partialTicks) {
         boolean flag = false;
 
         for (KeyBinding keybinding : this.options.keyBindings) {
-            if (keybinding.getKeyCode() != keybinding.getKeyCodeDefault()) {
-                flag = true;
-                break;
+            if (Controlling.isModernKeybindingInstalled && keybinding instanceof IKeyBinding) {
```

Comment:
This is the only place where the `instanceof' check is present. I think such a check should be present everywhere (just for safety). Also, please use modern Java cast syntax if possible.

Reply:
Should be good now